contrib/win32/win32compat/misc.c

```diff
@@ -1584,198 +1584,70 @@ get_sid(const char* name)
-		* its commandline without applying CRT or shell specific rules
-		*
-		* Ex.
-		* this works
-		*	cmd /c "c:\program files\sftp" -d
-		* this following wouldn't work in powershell, cygwin's or WSL bash unless we put
-		* some shell specific rules 
-		*	powershell -c "c:\program files\scp" -t
-		*	cygwin\bash -c "c:\program files\scp" -t
-		*	bash -c "c:\program files\scp" -t
-		* 
-		* for now, for all non-cmd shells, we launch scp and sftp-server directly -
-		*	shell -c "scp.exe -t"
-		* note that .exe extension and case matters for WSL bash
-		* note that double quotes matter for WSL and Cygwin bash, they dont matter for PS
-		*
-		* consequence - 
-		* for non-cmd shells - sftp and scp installation path is expected to be in machine wide PATH
-		* 
-		*/
-
-		int command_len;
-		const char *command_args = NULL;
-
-		if (!command)
-			break;
-		command_len = (int)strlen(command);
-		/*TODO - replace numbers below with readable compile time operators*/
-		if (command_len >= 13 && _memicmp(command, "internal-sftp", 13) == 0) {
-			command_type = CMD_SFTP;
-			command_args = command + 13;
-		} else if (command_len >= 11 && _memicmp(command, "sftp-server", 11) == 0) {
-			command_type = CMD_SFTP;
-
-			/* account for possible .exe extension */
-			if (command_len >= 15 && _memicmp(command + 11, ".exe", 4) == 0)
-				command_args = command + 15;
-			else
-				command_args = command + 11;
-		} else if (command_len >= 3 && _memicmp(command, "scp", 3) == 0) {
-			command_type = CMD_SCP;
-
-			/* account for possible .exe extension */
-			if (command_len >= 7 && _memicmp(command + 3, ".exe", 4) == 0)
-				command_args = command + 7;
-			else
-				command_args = command + 3;
-		}
-
-		if (command_type == CMD_OTHER)
-			break;
+	char *cmd_sp = NULL;
+	int len = 0;
 
-		len = 0;
-		len += progdir_len + 4; /* account for " around */
-		len += command_len + 4; /* account for possible .exe addition */
-
-		if ((cmd_sp = malloc(len)) == NULL) {
-			errno = ENOMEM;
-			goto done;
-		}
-
-		p = cmd_sp;
-
-		if ((shell_type == SH_CMD) || (shell_type == SH_CYGWIN)) {
-			CMDLINE_APPEND(p, "\"");
-			CMDLINE_APPEND(p, progdir);
-
-			if (command_type == CMD_SCP)
-				CMDLINE_APPEND(p, "\\scp.exe\"");
-			else
-				CMDLINE_APPEND(p, "\\sftp-server.exe\"");
-		} else {
-			if (command_type == CMD_SCP)
-				CMDLINE_APPEND(p, "scp.exe");
-			else
-				CMDLINE_APPEND(p, "sftp-server.exe");
-		}
-
-		if (shell_type == SH_CYGWIN) {
-			*p = '\0';
-			convertToForwardslash(cmd_sp);
-		}
+	if (!command)
+		return NULL;
 
-		CMDLINE_APPEND(p, command_args);
-		*p = '\0';
-		command = cmd_sp;
-	} while (0);
+	/*
+	* special case where incoming command needs to be adjusted 
+	* identify scp and sftp sessions
+	* we want to launch scp and sftp executables from the same binary directory
+	* that sshd is hosted in. This will facilitate hosting and evaluating
+	* multiple versions of OpenSSH at the same time.
+	*/
```

Comment:
looks like this function is not prepending the current binary location to command. are these comments obsolete ?

Reply:
1. we added current binary location to PATH
2. If we prepend the path, some shells, like powershell will require shell specific additional characters (& for powershell) to be able to run scp, sftp. powershell -c "& <full path to scp>"

Reply:
I will remove the comments